x/posts/internal/types/media_post.go

```diff
@@ -0,0 +1,260 @@
+package types
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type MediaPost struct {
+	TextPost
+	Medias PostMedias `json:"medias"`
+}
+
+func NewMediaPost(post TextPost, medias PostMedias) MediaPost {
+	return MediaPost{
+		TextPost: post,
+		Medias:   medias,
+	}
+}
+
+// String implements fmt.Stringer
+func (mp MediaPost) String() string {
+	bytes, err := json.Marshal(&mp)
+	if err != nil {
+		panic(err)
+	}
+
+	return string(bytes)
+}
+
+// GetID implements Post GetID
+func (mp MediaPost) GetID() PostID {
+	return mp.PostID
+}
+
+// GetParentID implements Post GetParentID
+func (mp MediaPost) GetParentID() PostID {
+	return mp.ParentID
+}
+
+func (mp MediaPost) SetMessage(message string) Post {
```

Comment:
Please add a method description using inline comments (`\\`) to this and all the below methods